R/dtree_cross_validation.R

```diff
@@ -0,0 +1,531 @@
+  overallAcc <- sum(thresholdedPredictions == actual)/length(actual)
+  PosAcc <- length(intersect(which(thresholdedPredictions == TRUE), actualPosIndic))/length(actualPosIndic)
+  NegAcc <- length(intersect(which(thresholdedPredictions == FALSE), actualNegIndic))/length(actualNegIndic)
+  precision <- nCorrectPos/nPredPos
+  recall <- nCorrectPos/nActualPos
+  F1 <- 2*(precision*recall)/(precision + recall)
+  tpr <- recall
+  rpp <- nPredPos/length(pred_prob)
+  lift <- tpr/rpp
+  fpr <- (nPredPos - nCorrectPos)/length(actualNegIndic)
+  pred <- ROCR::prediction(predictions = pred_prob, labels = actual)
+  auc <- ROCR::performance(pred, "auc")
+  auc <- unlist(auc@y.values)
+  data.frame(threshold = threshold, recall = recall, F1 = F1, lift = lift, Rate_Pos_Predictions = rpp, True_Pos_Rate = tpr, False_Pos_Rate = fpr, Precision = precision)
+}
+
+generateDataForPlots <- function(d, extras, config){
+  if (config$classification) {
+    if (length(extras$levels) == 2) {
+      thresholds <- seq(0, 1, 0.05)
+      plyr::ldply(thresholds, computeBinaryMetrics,
+            actual = ifelse(d$actual == extras$posClass, TRUE, FALSE),
+            pred_prob = d[[paste0('Score_', extras$posClass)]]
+      )
+    } else{
+      data.frame()
+    }
+  } else {
+    data.frame(response = d$response, actual = d$actual)
+  }
+}
+
+generateLabels <- function(plotData, config) {
+  trials <- c()
+  for (i in 1:length(unique(plotData$trial))) {
+    trials <- c(trials, paste0("Trial ", unique(plotData$trial))[i])
+  }
+  models <- c()
+  for (i in 1:length(unique(plotData$mid))) {
+    models <- c(models, paste0("Model ", unique(plotData$model))[i])
+  }
+  list(trials = trials, models = models)
+}
+
+#' @import ggplot2
+plotBinaryData <- function(plotData, config, modelNames) {
+  labels <- generateLabels(plotData, config)
+  modelVec <- modelNames[plotData$mid]
+  trialVec <- paste0('Trial ', plotData$trial)
+  plotData <- cbind(plotData, modelVec, trialVec)
+  liftdf <- data.frame(Rate_positive_predictions = plotData$Rate_Pos_Predictions, lift = plotData$lift, fold = paste0("Fold", plotData$fold),
+                       models = plotData$modelVec, trial = plotData$trialVec)
+  gaindf <- data.frame(Rate_positive_predictions = plotData$Rate_Pos_Predictions, True_Pos_Rate = plotData$True_Pos_Rate, fold = paste0("Fold", plotData$fold),
+                       models = plotData$modelVec, trial = plotData$trialVec)
+  prec_recalldf <- data.frame(recall = plotData$recall, precision = plotData$Precision, fold = paste0("Fold", plotData$fold),
+                              models = plotData$modelVec, trial = plotData$trialVec)
+  rocdf <- data.frame(False_Pos_Rate = plotData$False_Pos_Rate, True_Pos_Rate = plotData$True_Pos_Rate, fold = paste0("Fold", plotData$fold),
+                      models = plotData$modelVec, trial = plotData$trialVec)
+
+  liftPlotObj <- ggplot2::ggplot(data = liftdf, aes(x = Rate_positive_predictions, y = lift)) +
```

Comment:
You can get rid of all the warnings for `plotBinaryData` by replacing `aes` calls with `aes_string` calls and adding quotes to the arguments. For example, the line above would become

```r
liftPlotObj <- ggplot2::ggplot(data = liftdf, aes_string(x = "Rate_positive_predictions", y = "lift"))
```

I also think you can lose the `ggplot2::` here since it is very unlikely for function name conflicts with other packages.